jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/IO.java

```diff
@@ -736,13 +739,37 @@ public static File asFile(Object fileObject)
     {
         if (fileObject == null)
             return null;
-        if (fileObject instanceof File)
-            return (File)fileObject;
-        if (fileObject instanceof String)
-            return new File((String)fileObject);
-        if (fileObject instanceof Path)
-            return ((Path)fileObject).toFile();
+        if (fileObject instanceof File file)
+            return file;
+        if (fileObject instanceof Path path)
+            return path.toFile();
+        if (fileObject instanceof String str)
+        {
+            // attempt to support absolute, relative, partial, and URI syntaxes.
+            try (ResourceFactory.Closeable resourceFactory = ResourceFactory.closeable())
```

Comment:
This class did not depend on `Resource` before.

I would rather see all this `Resource` logic inside `Resource.asFile(Object)`, and keep this class simple and only depend on JDK classes.

Reply:
I only used Resource as it has all of the String (absolute, relative, partial, url, uri) to File manipulation already done.
I can implement it here in IO.asFile as well, just more work, more testing.

Reply:
Done, but not in the way you suggested.  Just removed the Resource stuff from IO.asFile instead.